packages/react-server-dom-webpack/src/__tests__/ReactFlightDOMBrowser-test.js

```diff
@@ -533,6 +533,42 @@ describe('ReactFlightDOMBrowser', () => {
     expect(container.innerHTML).toBe('{"foo":1}{"foo":1}');
   });
 
+  it('should resolve deduped references in maps used in client component props', async () => {
+    const ClientComponent = clientExports(function ClientComponent({
+      shared,
+      map,
+    }) {
+      return JSON.stringify({shared, map: Array.from(map)});
```

Comment:
Is `shared` supposed to be equal to `map.get(42) on the Client`? We should probably make that clear in the test.